libs/vkd3d/swapchain.c

```diff
@@ -1698,8 +1667,18 @@ static VkResult d3d12_swapchain_queue_present(struct d3d12_swapchain *swapchain,
 
     if (swapchain->vk_image_index == INVALID_VK_IMAGE_INDEX)
     {
-        if ((vr = d3d12_swapchain_acquire_next_vulkan_image(swapchain)) < 0)
+        if ((vr = vk_procs->vkAcquireNextImageKHR(vk_device, swapchain->vk_swapchain, UINT64_MAX,
+                swapchain->vk_acquire_semaphores[swapchain->frame_id], VK_NULL_HANDLE, &swapchain->vk_image_index)))
+        {
+            if (vr == VK_SUBOPTIMAL_KHR)
+            {
+                /* Suboptimal is still considered success, but we always want
+                 * to recreate swapchains in this case. */
+                return VK_ERROR_OUT_OF_DATE_KHR;
```

Comment:
Semaphore is left signalled here. Should submit a dummy submit to wait for the semaphore to get signaled, wait for a fence, and then destroy fence, and recreate semaphore.